src/Encrypt/Openssl.php

```diff
@@ -33,20 +33,34 @@
 use function substr;
 use function unpack;
 
-use const PHP_VERSION_ID;
-
 /**
  * Encryption adapter for openssl
+ *
+ * @psalm-type Options = array{
+ *     public?: string,
+ *     private?: string,
+ *     envelope?: string,
+ *     passphrase?: string,
+ *     package?: bool,
+ *     compression?: string|array{adapter: string},
+ * }
+ * @psalm-type CompressionOptions = array{
+ *     adapter: string,
+ * }&array<string, mixed>
+ * @final
```

Comment:
I think we should deprecate this contraption 🤔

Reply:
Just the OpenSSL adapter or the entire encrypt/decrypt filter? Happy to bin this PR and send in a fresh one that does that, along with disabling the broken OpenSSL test case.

Reply:
IMO the adapter at first: acting as an OpenSSL integrator is just asking for pain